### PR TITLE
Account Application - PELAGOS-3782

### PR DESCRIPTION
**Pull request by mickel1138**
Intentionally left commented out code in EventListener.php this construct does too much, which was convenient when we had to manually register every service, but now it's Autowired, and each service even if it's part of a base class, should ask for what they need at the appropriate time. So the commented out stuff is for reference, when other Listeners get added.
Yes, a listener still had to be registered in services,yml, we could refactor to EventSubscribers, but the mechanism (almost like a voter) is a bit different, and a refactor.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -43,5 +43,4 @@ HOSTNAME=host-env.tamucc.edu
 GOOGLE_MAPS_API_KEY=SECRET
 GOOGLE_ANALYTICS_ID=ID
 PELAGOS_READONLY_MESSAGE=MESSAGE
-MAINTENANCE_MODE_FILENAME="/var/maintenance.ini"
-ACCOUNT_PASSWORD_MAX_AGE=P180D
```
Comment (praneethpr) [inline .env]: Did you revert the changes for this file by mistake?

File: templates/template/ErrorMessage.html.twig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% extends "PelagosAppBundle:template:UI.html.twig" %}
```
Comment (praneethpr) [inline templates/template/ErrorMessage.html.twig]: Change the file path

File: src/Controller/UI/AccountController.php [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+    public function __construct(EntityHandler $entityHandler, ValidatorInterface $validator)
+    {
+        $this->entityHandler = $entityHandler;
+        $this->validator = $validator;
+    }
+
+    /**
+     * The index action.
+     *
+     * @Route("/account", methods={"GET"}, name="pelagos_app_ui_account_default")
```
Comment (praneethpr) [inline src/Controller/UI/AccountController.php]: I don't think we need to continue using the old names for the UI routes

File: src/Controller/UI/AccountController.php [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+        $password = new Password($request->request->get('password'));
+
+        // Set the creator for password.
+        $password->setCreator($person);
+
+        // Attach the password to the account.
+        try {
+            $account->setPassword(
+                $password,
+                ((bool) ($this->container->hasParameter('account_less_strict_password_rules')) and
```
Comment (praneethpr) [inline src/Controller/UI/AccountController.php]: Can remove container

File: src/Controller/UI/AccountController.php [hunk 5]
```diff
@@ -0,0 +1,514 @@
...
+        }
+
+        // Validate both Password and Account
+        $this->validateEntity($password, $validator);
+        $this->validateEntity($account, $validator);
+
+        $account = $this->entityHandler->update($account);
+
+        // Update LDAP
+        $this->get('pelagos.ldap')->updatePerson($person);
```
Comment (praneethpr) [inline src/Controller/UI/AccountController.php]: Here

File: src/Controller/UI/AccountController.php [hunk 6]
```diff
@@ -0,0 +1,514 @@
...
+
+            // Persist Account
+            $account = $this->entityHandler->create($account);
+
+            try {
+                // Try to add the person to LDAP.
+                $this->get('pelagos.ldap')->addPerson($person);
+            } catch (LdapException $exception) {
+                // If that fails, try to update the person in LDAP.
+                $this->get('pelagos.ldap')->updatePerson($person);
```
Comment (praneethpr) [inline src/Controller/UI/AccountController.php]: Here

File: src/Controller/UI/AccountController.php [hunk 7]
```diff
@@ -0,0 +1,514 @@
...
+            // Validate the entities.
+            $this->validateEntity($password);
+            $this->validateEntity($account);
+
+            // Persist Account
+            $account = $this->entityHandler->create($account);
+
+            try {
+                // Try to add the person to LDAP.
+                $this->get('pelagos.ldap')->addPerson($person);
```
Comment (praneethpr) [inline src/Controller/UI/AccountController.php]: Here

File: src/Controller/UI/AccountController.php [hunk 8]
```diff
@@ -0,0 +1,514 @@
...
+            }
+
+            // Validate the entities.
+            $this->validateEntity($password, $validator);
+            $this->validateEntity($account, $validator);
+
+            // Persist Account
+            $account = $this->entityHandler->update($account);
+
+            $this->get('pelagos.ldap')->updatePerson($person);
```
Comment (praneethpr) [inline src/Controller/UI/AccountController.php]: Inject this as a service? and change usages below

File: templates/Account/Account.html.twig [hunk 9]
```diff
@@ -0,0 +1,41 @@
+{% extends "template/UI.html.twig" %}
+
+{% block javascripts %}
+    {{ parent() }}
+    
+    <script type="text/javascript" src="{{ asset('build/js/Account.js') }}"></script>
```
Comment (praneethpr) [inline templates/Account/Account.html.twig]: Should remove build from the path?

File: src/Controller/UI/AccountController.php (hunk 3 above)
Comment (mickel1138) [reply src/Controller/UI/AccountController.php]: In the future we should change them, i need them for now.
